sharness/t0020-service-basic-commands.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+test_description="Test service startup and init functionality"
+
+. lib/test-lib.sh
+
+test_expect_success "ipfs setup" '
+    test_ipfs_init &&
+    test_have_prereq IPFS_INIT
```

Comment:
How I'd do this:

```
. lib/test-lib.sh

test_ipfs_init
test_cluster_init

test_expect_success "prerequisites" '
  test_have_prereq IPFS_INIT
  test_have_prereq CLUSTER_INIT
  ....
  '
```

Then, in every other test, only run if the prerequisite is set:

```
test_expect_success IPFS_INIT,CLUSTER_INIT "ipfs cluster setup" '...'
```

Please do this this way in every test file. Right now tests are running even when prerequisites aren't there so there will be many failures if ipfs fails to start, making it hard to see. It should just fail the prerrequisites test and skip every other test.